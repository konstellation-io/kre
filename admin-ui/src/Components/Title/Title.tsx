import React, { FC } from 'react';
import styles from './Title.module.scss';

type Props = {
  children: (string | JSX.Element)[];
};
const Title: FC<Props> = ({ children }) => (
  <div className={styles.header}>
    <div className={styles.title}>{children}</div>
  </div>
);

export default Title;
