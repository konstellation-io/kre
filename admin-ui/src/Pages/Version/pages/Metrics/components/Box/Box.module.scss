@import 'Styles/mixins';
@import 'Styles/colors';
@import 'Styles/shadows';

.container {
  @include shadow(4);

  position: relative;
  width: 100%;
  height: 100%;
  border-radius: 4px;
  background-color: $bg-box;
}

.content {
  display: flex;
  flex-direction: column;
  width: 100%;
  height: 100%;
  padding: 13px 18px;
  transition: all ease 2s;

  p {
    margin: 0;
  }
}

.title {
  @include font-body;
  color: font-color(light);
}

.infoNumber {
  color: font-color(highlight);
  font-weight: 400;

  &.regular {
    @include font-title;
  }
  &.big {
    @include font-headline;
  }
}

.expandIcon {
  position: absolute;
  top: 16px;
  right: 16px;
  cursor: pointer;
  transition: color ease 0.4s;
  z-index: 1;

  &:hover {
    color: font-color(light);
  }
}
