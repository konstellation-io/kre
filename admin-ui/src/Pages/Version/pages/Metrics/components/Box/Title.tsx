import React from 'react';
import styles from './Box.module.scss';

type Props = {
  text: string;
};
function Title({ text }: Props) {
  return <p className={styles.title}>{text}</p>;
}

export default Title;
