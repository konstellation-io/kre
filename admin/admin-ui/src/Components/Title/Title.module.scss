@import 'Styles/colors';
@import 'Styles/mixins';

.header {
  padding: 3 * $grid-unit;
  background-color: palette(base, 600);
}

.title {
  @include font-headline;
  color: font-color(lowlight);
}
