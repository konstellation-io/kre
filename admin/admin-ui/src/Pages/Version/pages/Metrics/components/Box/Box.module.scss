@import 'Styles/mixins';
@import 'Styles/colors';
@import 'Styles/shadows';

.container {
  @include shadow(4);

  position: relative;
  width: 100%;
  height: 100%;
  border-radius: 4px;
  background-color: $bg-box;
}

.content {
  display: flex;
  flex-direction: column;
  width: 100%;
  height: 100%;
  padding: 13px 18px;
  transition: all ease 2s;

  p {
    margin: 0;
  }
}

.title {
  @include font-body;
  color: font-color(light);
}

.infoNumber {
  color: font-color(highlight);
  font-weight: 400;

  &.regular {
    @include font-title;
  }
  &.big {
    @include font-headline;
  }
}

.expandIcon {
  position: absolute;
  top: 16px;
  right: 16px;
  z-index: 1;
}

.help {
  position: absolute;
  top: 16px;
  right: 58px;
  z-index: 1;
}

.modalContainer {
  @include font-small;
  font-weight: 500;

  position: absolute;
  background-color: palette(base, 400);
  z-index: 1;
  transform: translate(-100%, -50%);
  border-radius: 0.5 * $grid-unit;

  &.visible {
    @include shadow(4);

    &:after {
      content: '';
      border: solid 6px palette(base, 400);
      border-left-color: transparent;
      border-bottom-color: transparent;
      transform: rotate(45deg);
      position: absolute;
      right: -5px;
      top: calc(50% - 7px);
    }

    padding: 1.5 * $grid-unit 2 * $grid-unit;
    width: 45 * $grid-unit;
  }

  .modal {
    position: absolute;
    top: 0;
  }
}
