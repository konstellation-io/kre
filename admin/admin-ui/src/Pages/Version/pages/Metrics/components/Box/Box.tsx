import React, { FunctionComponent, ReactElement } from 'react';
import styles from './Box.module.scss';

type Props = {
  children?: ReactElement | ReactElement[];
};
const Box: FunctionComponent<Props> = ({ children }) => {
  return (
    <section className={styles.container}>
      <div className={styles.content}>{children}</div>
    </section>
  );
};

export default Box;
