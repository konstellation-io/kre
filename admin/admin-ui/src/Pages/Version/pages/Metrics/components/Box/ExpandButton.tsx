import React, { MouseEvent } from 'react';

import Button from 'Components/Button/Button';
import IconExpand from '@material-ui/icons/Fullscreen';
import styles from './Box.module.scss';

type Props = {
  onClick: (e: MouseEvent<HTMLDivElement>) => void;
};
function ExpandButton({ onClick }: Props) {
  return (
    <Button
      className={styles.expandIcon}
      Icon={IconExpand}
      iconSize="icon-regular"
      label=""
      onClick={onClick}
    />
  );
}

export default ExpandButton;
