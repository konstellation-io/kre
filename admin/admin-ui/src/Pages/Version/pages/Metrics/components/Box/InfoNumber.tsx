import React from 'react';
import cx from 'classnames';
import styles from './Box.module.scss';

export enum Sizes {
  REGULAR = 'regular',
  BIG = 'big'
}

type Props = {
  text: string;
  size?: Sizes;
};
function InfoNumber({ text, size = Sizes.REGULAR }: Props) {
  return <p className={cx(styles.infoNumber, styles[size])}>{text}</p>;
}

export default InfoNumber;
