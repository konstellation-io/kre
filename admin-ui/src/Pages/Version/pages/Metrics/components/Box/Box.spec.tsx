import React from 'react';
import { shallow } from 'enzyme';
import Box from './Box';
import ExpandButton from './ExpandButton';
import InfoNumber from './InfoNumber';
import Title from './Title';

let wrapper;

describe('Box', () => {
  beforeEach(() => {
    wrapper = shallow(
      <Box>
        <div>Some component</div>
      </Box>
    );
  });

  it('matches snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});

describe('ExpandButton', () => {
  beforeEach(() => {
    wrapper = shallow(<ExpandButton onClick={jest.fn()} />);
  });

  it('matches snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});

describe('InfoNumber', () => {
  beforeEach(() => {
    wrapper = shallow(<InfoNumber text="some text" size={'regular'} />);
  });

  it('matches snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});

describe('Title', () => {
  beforeEach(() => {
    wrapper = shallow(<Title text="some text" />);
  });

  it('matches snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
