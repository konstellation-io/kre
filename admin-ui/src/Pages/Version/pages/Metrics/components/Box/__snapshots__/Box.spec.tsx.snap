// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Box matches snapshot 1`] = `
<section
  className="container"
>
  <div
    className="content"
  >
    <div>
      Some component
    </div>
  </div>
</section>
`;

exports[`ExpandButton matches snapshot 1`] = `
<div
  className="expandIcon"
  onClick={[MockFunction]}
>
  <FullscreenIcon
    className="icon-regular"
  />
</div>
`;

exports[`InfoNumber matches snapshot 1`] = `
<p
  className="infoNumber regular"
>
  some text
</p>
`;

exports[`Title matches snapshot 1`] = `
<p
  className="title"
>
  some text
</p>
`;
